source-code/sdk-js/src/adapter-sveltekit/vite-plugin/transforms/refactoring.md

```diff
@@ -113,6 +113,144 @@ for markup block:
 
 # Flows
 
+## `Javascript` and `Svelte`
+
+There is some stuff that has to be done both in javascript AND every svelte file:
+
+Possible implementation:
+
+```ts
+function wrapAndAddParameters(
+	ast,
+	config,
+	wrappers: (
+		| "initHandleWrapper"
+		| "initRootLayoutServerLoadWrapper"
+		| "initServerLoadWrapper"
+		| "initActionWrapper"
+		| "initRequestHandlerWrapper"
+		| "initRootLayoutLoadWrapper"
+		| "initLoadWrapper"
+		| "initRootPageLoadWrapper"
+	)[],
+) {
+	wrappers.forEach((wrapper) => {
+		let identifier
+		let wrapperAst
+		switch (wrapper) {
+			case "initHandleWrapper":
+				identifiers = ["handle"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			case "initServerLoadWrapper":
+			case "initRootLayoutServerLoadWrapper":
+				identifiers = ["load"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			case "initRootLayoutLoadWrapper":
+			case "initRootPageLoadWrapper":
+			case "initLoadWrapper":
+				identifiers = ["load"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/shared"
+				break
+			case "initRequestHandlerWrapper":
+				identifiers = ["GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			case "initActionWrapper":
+				identifiers = ["actions"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			default:
+				break
+		}
+		// Wrap the specified functions
+		const wasWrapped = definitions(ast, ...identifiers)
+			.wrap(wrapperAst)
+			.successful()
+		if (wasWrapped) imports(ast, importFrom).add(wrapper)
+
+		// Add imports
+	})
+	const aliases = imports(ast, "@inlang/sdk-js").getAliases()
+	// Ensure that function parameters are correct
+	contexts(ast, ...aliases.map(({ aliasN }) => aliasN))
+		.isFunction()
+		.merge(
+			parse(
+				`function(_, {${aliases
+					.map(({ exportN, aliasN }) => `${exportN}:${aliasN}`)
```

Comment:
can we create a helper function that only outputs an alias if the import got renamed?

Reply:
I am not sure that I understand - which specific line are you talking about?

Reply:
This should not be a problem any more as I changed the strategy:

1. Remove sdk imports first
2. Replace all modules from the imports with placeholders
3. Work with these placeholders